[MOBILE-1650] Add badge methods for iOS

rlepinski opened the pull request:
### What do these changes do?
Adds badge methods and properties for iOS.

### How did you verify these changes?
Ran sample, verified setters/getters.

BrianBatchelder commented on ios/Classes/TiAirshipModule.m:
```diff
@@ -118,6 +118,26 @@ - (void)setUserNotificationsEnabled:(id)value {
+}
+
+- (void)setIsAutoBadgeEnabled:(id)args {
+    [UAirship push].autobadgeEnabled = [TiUtils boolValue:args def:YES];
```
Just wondering if the default should be the current value, i.e. a no-op.

BrianBatchelder commented on ios/Classes/TiAirshipModule.m:
```diff
@@ -118,6 +118,26 @@ - (void)setUserNotificationsEnabled:(id)value {
+}
+
+- (void)setBadgeNumber:(id)args {
+    [UAirship push].badgeNumber = [TiUtils boolValue:args def:YES];
```
Shouldn't this be an int rather than a bool?

rlepinski replied:
yeah, i was testing with 1 so i didnt notice...

marc-scig commented on ios/Classes/TiAirshipModule.m:
```diff
@@ -118,6 +118,28 @@ - (void)setUserNotificationsEnabled:(id)value {
+
+- (void)setIsAutoBadgeEnabled:(id)args {
+    [UAirship push].autobadgeEnabled = [TiUtils boolValue:args
+                                                      def:[UAirship push].autobadgeEnabled];
```
what are the conditions in which args could be nil? I'm wondering if maybe we want to do a similar thing in setUserNotificationsEnabled, which currently has YES as the default value.

marc-scig commented on ios/Classes/TiAirshipModule.m:
```diff
@@ -118,6 +118,28 @@ - (void)setUserNotificationsEnabled:(id)value {
+}
+
+- (void)setBadgeNumber:(id)args {
+    [UAirship push].badgeNumber = [TiUtils numberFromObject:args].integerValue;
```
Does this need a default value? Or likewise could the other one be done with:

`[TiUtils numberFromObject:args].boolValue`

rlepinski commented:
Cleaned up arg handling and fixed the custom event method to accept an object instead of being forced to jsonify. I left handling both to keep backwards compatibility.

marc-scig commented on ios/Classes/TiAirshipModule.m:
```diff
@@ -165,19 +187,33 @@ - (void)associateIdentifier:(id)args {
+            return;
+        }
+    } else {
+        ENSURE_SINGLE_ARG(args, NSDictionary);
```
I'm a little confused by this, should it be `ENSURE_SINGLE_ARG(args[0], NSDictionary)`?  How can args have an index and be a dictionary at the same time?

rlepinski replied:
Magic. The macro redefines the variable
